streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -37,7 +37,6 @@
 
 public class ProcessorNode<KIn, VIn, KOut, VOut> {
 
-    // TODO: 'children' can be removed when #forward() via index is removed
```

Comment:
`children` is actually used again in the meantime and we should keep it.

(It's of course somewhat redundant to `childByName` but is also preserves the order of the nodes -- using a `TreeMap` might work however, as we return a `List` of child name form one method it would imply extra work to extract the keys...)

Reply:
Fine with me

Reply:
Cool. We can still change it later as it's only internal stuff if we feel the need.